src/DynamoCore/Models/NodeModel.cs

```diff
@@ -614,8 +616,50 @@ public virtual ProtoCore.Type GetTypeHintForOutput(int index)
         {
              return ProtoCore.TypeSystem.BuildPrimitiveTypeObject(ProtoCore.PrimitiveType.kTypeVar);
         }
-        #endregion
 
+      
+        /// <summary>
+        /// Gets or sets a value indicating whether this node is explictly frozen.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this node is frozen; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsFrozen
+        {
+            get
+            {
+                return isFrozen;
+            }
+
+            set
+            {
+                isFrozen = value;                
+                RaisePropertyChanged("IsFrozen");
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this node is implictly frozen.
```

Comment:
Ok. We can discuss.. Explicit freezing is that you make the node to freeze. Implicit is because the input node is frozen, all the downstream nodes are frozen.